Tasks/AzureIoTEdge/telemetry.ts

```diff
@@ -0,0 +1,38 @@
+import * as appInsights from 'applicationinsights';
```

Comment:
Are we pushing telemetry data? Typically all in-built task pushes data to Kusto using ##vso[telemetry.publish. Also I am not sure if it is ok in on-premise TFS.

Reply:
Yes. We're sending some telemetry data to Application Insights for usage statistics. We have a [statement](https://github.com/Microsoft/azure-iot-edge-vsts-extension#datatelemetry) for the telemetry, and user can disable the telemetry by setting variable `DISABLE_TELEMETRY` to `true`.

Reply:
In on-premise server, we push telemetry data only when user gives consent.

Reply:
I have several questions about telemetry on on-premise server

1. In on-premise server, typically you push data to Kusto. Does that need user consent?
2. Can I detect if the task is running in on-premise server or DevOps? If yes, I think we can choose to push to Application Insights when the server is DevOps (Maybe push to Kusto for on-premise server)
3. If we change to logic to push all telemetry data to Kusto, how can our team query the data and use it for analytics
Thanks

Reply:
And our telemetry to AI will not persist(last for 2 days), then a back-end service will route these data to Kusto (under Visual Studio org). So in concept we also use Kusto to record telemetry.

Reply:
two things here:
1. We should try and push telemetry only for hosted AzureDevops . I will get back on what parameter you can check to make the telemetry call
2. Recommend you also push to kusto as well as all inhouse tasks telemetry is gathered here

Reply:
Also, assuming failure to log telemetry does not fail the task

Reply:
use tl.getVariable("System.ServerType") and check it for 'hosted' value 
Reference: https://github.com/Microsoft/azure-pipelines-tasks/blob/01c1662494fb46710c1100a35d1ccb60e1126cdb/Tasks/Common/packaging-common/locationUtilities.ts